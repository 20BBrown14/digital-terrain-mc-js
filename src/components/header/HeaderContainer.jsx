import React from 'react';
import PropTypes from 'prop-types';
import selectedNavKeyPropTypes from '../../proptypes/selectedNavKeyPropTypes';
import HeaderView from './HeaderView';

const propTypes = {
  /* The page to load base on the requested url from server */
  selectedNavKey: selectedNavKeyPropTypes.isRequired,
  /* Function for handling menu item click */
  handleNavMenuClick: PropTypes.func.isRequired,
};

/**
 * Header container. Renders Header view.
 */
class HeaderContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = { showTitle: true };

    this.updateShowTitle = this.updateShowTitle.bind(this);
  }

  componentDidMount() {
    this.updateShowTitle();
    window.addEventListener('resize', this.updateShowTitle);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateShowTitle);
  }

  updateShowTitle() {
    this.setState({ showTitle: (window.innerWidth > 442 || window.innerWidth < 388) });
  }

  render() {
    const { selectedNavKey, handleNavMenuClick } = this.props;
    const { showTitle } = this.state;
    return (
      <HeaderView
        selectedNavKey={selectedNavKey}
        handleNavMenuClick={handleNavMenuClick}
        showTitle={showTitle}
      />
    );
  }
}

HeaderContainer.propTypes = propTypes;

export default HeaderContainer;
