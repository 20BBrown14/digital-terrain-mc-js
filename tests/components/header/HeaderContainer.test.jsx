import React from 'react';
import HeaderContainer from '../../../src/components/header/HeaderContainer';

describe('HeaderContainer', () => {
  let container;
  beforeEach(() => {
    container = shallow(<HeaderContainer />);
  });

  it('renders a default view', () => {
    container = render(<HeaderContainer />);
    expect(container).toMatchSnapshot();
  });

  it('it has a view', () => {
    const view = container.find('HeaderView');
    expect(view).toHaveLength(1);
  });
});
