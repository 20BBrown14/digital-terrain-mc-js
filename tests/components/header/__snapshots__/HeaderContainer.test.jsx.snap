// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer renders a default view 1`] = `
<div>
  <ul
    class="ant-menu header-menu ant-menu-light ant-menu-root ant-menu-horizontal"
    role="menu"
  >
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      aria-disabled="true"
      class="ant-menu-item header-icon-menu-item ant-menu-item-only-child ant-menu-item-disabled"
      role="menuitem"
    >
      <img
        alt="loading..."
        class="header-icon"
        src="test-file-stub"
      />
      <span />
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item header-title header-non-highlighted-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Digital Terrain
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
      role="menuitem"
    >
      Home
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Gallery
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Map
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Rules
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Server Information
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      Apply
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      About us
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="display:none"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      class="ant-menu-item ant-menu-item-only-child"
      role="menuitem"
    >
      <a
        href="https://discord.com/api/oauth2/authorize?client_id=693123471210709072&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Floggedin&response_type=code&scope=identify%20guilds"
      >
        Login
      </a>
    </li>
    <li
      class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
      role="menuitem"
      style="visibility:hidden;position:absolute"
    >
      <div
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-menu-submenu-title"
        role="button"
      >
        <span>
          ···
        </span>
        <i
          class="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
</div>
`;
